Acceptance test fail on "typing.py"

The followint test: ``pytest -m acceptance -k "test_libmodule[typing.py]"``

Fail with the following stacktrace:
```
    def child_sequence(self, child):
        """Search for the sequence that contains this child.
    
        :param child: The child node to search sequences for.
        :type child: NodeNG
    
        :returns: The sequence containing the given child node.
        :rtype: iterable(NodeNG)
    
        :raises AstroidError: If no sequence could be found that contains
            the given child.
        """
        for field in self._astroid_fields:
            node_or_sequence = getattr(self, field)
            if node_or_sequence is child:
                return [node_or_sequence]
            # /!\ compiler.ast Nodes have an __iter__ walking over child nodes
            if (
                isinstance(node_or_sequence, (tuple, list))
                and child in node_or_sequence
            ):
                return node_or_sequence
    
        msg = "Could not find %s in %s's children"
>       raise exceptions.AstroidError(msg % (repr(child), repr(self)))
E       astroid.exceptions.AstroidError: Could not find 
<ClassDef.Hashable_typing l.0 at 0x7fa3ceac21f0> in <Module._collections_abc l.0 at 0x7fa3cea020a0>'s children
```

--- a/ChangeLog
+++ b/ChangeLog
@@ -26,6 +26,10 @@ Release date: Undefined
 
   Closes #4301
 
+* Updated ``astroid`` to 2.5.3
+
+  Closes #2822, #4206, #4284
+
 
 What's New in Pylint 2.7.5?
 ===========================

--- a/doc/whatsnew/2.8.rst
+++ b/doc/whatsnew/2.8.rst
@@ -26,3 +26,5 @@ Other Changes
 
 * The 'doc' extra-require has been removed. ``pylint.version`` is now ``pylint.__version__`` and ``__pkginfo__`` does
   not contain the package metadata anymore.
+
+* Updated ``astroid`` to 2.5.3

--- a/requirements_test_min.txt
+++ b/requirements_test_min.txt
@@ -1,4 +1,4 @@
 -e .
-astroid==2.5.2  # Pinned to a specific version for tests
+astroid==2.5.3  # Pinned to a specific version for tests
 pytest~=6.2
 pytest-benchmark~=3.2

--- a/setup.cfg
+++ b/setup.cfg
@@ -42,7 +42,7 @@ project_urls =
 [options]
 packages = find:
 install_requires =
-    astroid>=2.5.2,<2.7
+    astroid>=2.5.3,<2.7
     isort>=4.2.5,<6
     mccabe>=0.6,<0.7
     toml>=0.7.1

--- a/tests/functional/n/name/name_styles.py
+++ b/tests/functional/n/name/name_styles.py
@@ -151,7 +151,7 @@ class FooEnum(Enum):
     bad_enum_name = 2  # [invalid-name]
 
 class Bar:
-    """Class with class constants annotated with ClassVar."""
+    """Class with class variables annotated with ClassVar."""
     CLASS_CONST: ClassVar[int] = 42
     CLASS_CONST2: ClassVar = "const"
     variable: ClassVar[str] = "invalid name"

--- a/tests/functional/t/typing_generic.py
+++ b/tests/functional/t/typing_generic.py
@@ -1,0 +1,14 @@
+# pylint: disable=missing-docstring,invalid-name,too-few-public-methods
+
+# https://github.com/PyCQA/pylint/issues/2822
+# Base should be subscriptable, even with ABCMeta as metaclass
+from abc import ABCMeta
+from typing import Generic, TypeVar
+
+T = TypeVar("T")
+
+class Base(Generic[T], metaclass=ABCMeta):
+    """Base"""
+
+class Impl(Base[str]):
+    """Impl"""

--- a/tests/functional/t/typing_generic.rc
+++ b/tests/functional/t/typing_generic.rc
@@ -1,0 +1,2 @@
+[testoptions]
+min_pyver=3.7

